accessible/clipboard.service.js

```diff
@@ -0,0 +1,110 @@
+      this.markedConnection_ = null;
+    },
+    cut: function(block) {
+      this.clipboardBlockXml_ = Blockly.Xml.blockToDom(block);
+      this.clipboardBlockSuperiorConnection_ = block.outputConnection ||
+          block.previousConnection;
+      this.clipboardBlockNextConnection_ = block.nextConnection;
+      block.dispose(true);
+      blocklyApp.debug && console.log('cut');
+    },
+    copy: function(block) {
+      this.clipboardBlockXml_ = Blockly.Xml.blockToDom(block);
+      this.clipboardBlockSuperiorConnection_ = block.outputConnection ||
+          block.previousConnection;
+      this.clipboardBlockNextConnection_ = block.nextConnection;
+      blocklyApp.debug && console.log('copy');
+    },
+    paste: function(connection) {
+      var blockOnProperWorkspace = Blockly.Xml.domToBlock(blocklyApp.workspace,
+          this.clipboardBlockXml_);
+      switch (connection.type) {
+        case Blockly.NEXT_STATEMENT:
+          connection.connect(blockOnProperWorkspace.previousConnection);
+          break;
+        case Blockly.PREVIOUS_STATEMENT:
+          connection.connect(blockOnProperWorkspace.nextConnection);
+          break;
+        default:
+          connection.connect(blockOnProperWorkspace.outputConnection);
+      }
+      blocklyApp.debug && console.log('paste');
+    },
+    pasteToMarkedConnection: function(block) {
+      var xml = Blockly.Xml.blockToDom(block);
+      var blockOnProperWorkspace =
+          Blockly.Xml.domToBlock(blocklyApp.workspace, xml);
+      this.markedConnection_.connect(
+          blockOnProperWorkspace.outputConnection ||
+          blockOnProperWorkspace.previousConnection);
+      blocklyApp.debug && console.log('paste to mark connection');
+    },
+    markConnection: function(connection) {
+      this.markedConnection_ = connection;
+      blocklyApp.debug && console.log('mark connection');
+    },
+    isCompatibleWithConnection_: function(blockConnection, connection) {
+      // the connection and blockConnection exist
+      if (!connection || !blockConnection) {
+        return false;
+      }
+
+      // the types match and it's the right kind of connection
+      var result = Blockly.OPPOSITE_TYPE[blockConnection.type] ==
+          connection.type && connection.checkType_(blockConnection);
+
+      if (blocklyApp.debug) {
+        if (result) {
+          console.log('Blocks should be connected');
+        } else {
+          console.log('Blocks not connected');
+        }
+      }
+      return result;
+    },
+    isBlockCompatibleWithMarkedConnection: function(block) {
+      var blockConnection = block.outputConnection || block.previousConnection;
+      return this.markedConnection_ &&
+          this.markedConnection_.sourceBlock_.workspace != null &&
+          this.isCompatibleWithConnection_(
+              blockConnection, this.markedConnection_);
+    },
+    isConnectionCompatibleWithClipboard: function(connection) {
```

Comment:
Noted, but I probably won't change the name. I don't think it's work the extra word in the function title.